freqtrade/strategy/interface.py

```diff
@@ -504,18 +504,20 @@ def get_signal(self, pair: str, timeframe: str, dataframe: DataFrame) -> Tuple[b
                 'Outdated history for pair %s. Last tick is %s minutes old',
                 pair, int((arrow.utcnow() - latest_date).total_seconds() // 60)
             )
-            return False, False
+            return False, False, ''
 
-        (buy, sell) = latest[SignalType.BUY.value] == 1, latest[SignalType.SELL.value] == 1
+        (buy, sell, buy_signal_name) = latest[SignalType.BUY.value] == 1,\
+            latest[SignalType.SELL.value] == 1,\
+            latest.get(SignalNameType.BUY_SIGNAL_NAME.value, '')
```

Comment:
I don't think this is valid syntax (not for this object-type, anyway).

`latest` is not a dictionary (which would offer .get()`) - but a pandas Series.
As such - you'll not be able to use it this way, but you'll have to explicitly check if the signal column is part of the dataframe - and only then assign it with it's value.

Reply:
pandas has ```.get()```
https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.get.html

I tried and everything ok. I will add test case to prove this